### PR TITLE
♻️✨ Improve logging output in case of error when scheduling tasks

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- the dask clients pool was reporting invalid errors with a valid cause in the client, thus generating confusion
- the traceback of the error was missing making it difficult to narrow down the issue

This PR shall help in fixing this.

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]
:ok_hand:

File: services/director-v2/src/simcore_service_director_v2/modules/dask_clients_pool.py [hunk 1]
```diff
@@ -102,30 +99,20 @@ async def _concurently_safe_acquire_client() -> DaskClient:
...
-                await dask_client.delete()
-            logger.error(
-                "could not create/access dask computational cluster %s",
-                json_dumps(cluster),
-            )
             raise DaskClientAcquisisitonError(cluster=cluster, error=exc) from exc
+        else:
+            try:
+                yield dask_client
+            except (
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/dask_clients_pool.py:107]: so now every other exception is just "let-go" ... who is handling it then?

**Review by mguidon — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/base_scheduler.py [hunk 2]
```diff
@@ -434,8 +434,9 @@ async def _schedule_next_tasks(
                 )
             elif isinstance(r, Exception):
                 logger.error(
-                    "Unexpected error happened when scheduling task due to following error %s",
+                    "Unexpected error happened when scheduling task due to following error %s\n%s",
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/base_scheduler.py:437]: good point

File: services/director-v2/src/simcore_service_director_v2/modules/dask_clients_pool.py (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/dask_clients_pool.py:107]: the scheduler

**Pull request merged**